packages/cli/src/metadataGeneration/typeResolver.ts

```diff
@@ -420,6 +420,20 @@ export class TypeResolver {
         return stringMetaType;
       }
 
+      if (typeReference.typeName.text === 'Record' && typeReference.typeArguments?.length === 2) {
+        const indexType = new TypeResolver(typeReference.typeArguments[0], this.current, this.parentNode, this.context).resolve();
+        if (indexType.dataType === 'string') {
+          const valueType = new TypeResolver(typeReference.typeArguments[1], this.current, this.parentNode, this.context).resolve();
```

Comment:
In case we are passing from parent (GenericModel<T> has items: Record<string, T>) can you handle these without amending the context? (IIRC that should be created on processing the `GenericModel<T>`, but it would also be a good test probably.

Reply:
How would I go about doing that? Should I be creating a new temporary context?

I'm not familiar with the codebase so I'm not sure what the context is here, or what purpose it serves, so I just followed the structure of other cases in this method.

Reply:
This should be handled for you at this point already.

Reply:
Sorry, I'm not sure what you mean.